### PR TITLE
MULE-9205: XmppMessageDispatcher doesn't reconnect

### PR DESCRIPTION
**Pull request by elrodro83**

### CODE REVIEW

File: transports/xmpp/src/main/java/org/mule/transport/xmpp/XmppMessageDispatcher.java [hunk 1]
```diff
@@ -80,6 +80,12 @@ protected MuleMessage doSend(MuleEvent event) throws Exception
 
     protected void sendMessage(MuleEvent event) throws Exception
     {
+        // Handle session closing by the server
```
Comment (marianogonzalez) [inline transports/xmpp/src/main/java/org/mule/transport/xmpp/XmppMessageDispatcher.java]: what about the message receiver?

File: transports/xmpp/src/main/java/org/mule/transport/xmpp/XmppMessageDispatcher.java [hunk 2]
```diff
@@ -80,6 +80,12 @@ protected MuleMessage doSend(MuleEvent event) throws Exception
...
+        if (!xmppConnector.getXmppConnection().isConnected())
+        {
+            xmppConnector.getXmppConnection().connect();
```
Comment (marianogonzalez) [inline transports/xmpp/src/main/java/org/mule/transport/xmpp/XmppMessageDispatcher.java]: what if this fails?

File: transports/xmpp/src/main/java/org/mule/transport/xmpp/XmppMessageDispatcher.java (hunk 1 above)
Comment (elrodro83) [inline transports/xmpp/src/main/java/org/mule/transport/xmpp/XmppMessageDispatcher.java]: The receiver uses a Smack listener. For that case, Smack reconnects automatically.

File: transports/xmpp/src/main/java/org/mule/transport/xmpp/XmppMessageDispatcher.java (hunk 2 above)
Comment (elrodro83) [inline transports/xmpp/src/main/java/org/mule/transport/xmpp/XmppMessageDispatcher.java]: An exception will be thrown. If the remote server is down or rejects the connection, Smack's reconnection mechanism will kick in.

**Pull request merged**